wf-galleon/WFLY-19021-Stability_In_Provisioning.adoc

```diff
@@ -0,0 +1,165 @@
+
+* wildfly-ee-galleon-pack: `experimental`
+* wildfly-galleon-pack: `experimental`
+* wildfly-preview-feature-pack: `experimental`
+
+=== Requirements at provisioning time
+
+Provisioning time follows some rules to properly handle stability and avoid features/packages of wrong stability to be provisioned.
+
+* The feature-pack contained minimum stability level is used to constrain the set of provisioned features. Any features/packages at a lower stability level contained in the 
+feature-pack (if any), are not provisioned.
+* By default, the minimum stability level of each feature-pack is used when provisioning its own content. 
+* Each feature-pack present in the provisioning configuration can have a different minimum stability level. 
+The minimum stability of each feature-pack applies to its own content and doesn't leak into other feature-packs.
+* A user can specify a stability-level, by the mean of the `stability-level` Galleon option. 
+This stability level is used to constrain the stability level of all provisioned features/packages. If the specified level is lower than a feature-pack minimum stability, 
+the feature-pack minimum stability is used.
+
+For example, a provisioning configuration containing 3 feature-packs:
+
+* A, minimum stability default
+* B, minimum stability community
+* C, minimum stability experimental
+
+If no stability level is specified:
+
+* `default` features/packages of A are provisioned
+* `community` and `default` features/packages of B are provisioned
+* `experimental`, `preview`, `community` and `default` features/packages of C are provisioned
+
+If the stability level `default` is specified:
+
+* `default` features/packages of A are provisioned
+* `default` features/packages of B are provisioned
+* `default` features/packages of C are provisioned
+
+If the stability level `experimental` is specified:
+
+* `default` features/packages of A are provisioned
+* `community` and `default` features/packages of B are provisioned
+* `experimental`, `preview`, `community` and `default` features/packages of C are provisioned
+
+==== Lowest stability level needed by WildFly Galleon Plugins
+
+At provisioning time, when WildFly Galleon plugin generates the server configurations, 
+a WildFly embedded server is started. This server is started with a stability level computed in the following way:
+
+* If a stability level has been specified by the user, this level is used.
+* If no stability level has been specified, the lowest minimum stability level of all feature-packs present in the provisioning is used.
+
+=== Provisioning tooling requirement
+
+* All Galleon based tooling should let users specify the `stability-level` option.
+* In addition, for WildFly Glow, information about the features/packages that would be not provisioned for a given stability level 
+is displayed to the user.
+
+=== Nice-to-Have Requirements
+
+* NONE
+
+=== Non-Requirements
+
+* NONE
```

Comment:
We need a non-requirement related to stability levels for Galleon constructs that are not directly tied to management model metadata and JBoss Modules modules. Perhaps

* Recording and utilizing stability information in Galleon elements (e.g. configs, layers, feature-groups, features, packages) that are not generated by reading the server management model metadata (i.e. the 'read-feature-description' management operation) or reading 'module.xml' files that are made available during a feature pack build.'

This may be too broad though, as I suspect manually adding stability metadata to a feature-spec.xml or package.xml would work fine. That is I assume when we generate feature-spec.xml and package.xml we are recording stability metadata, so there's no reason humans couldn't add it for non-generated files.

In that case, perhaps:

* Recording and utilizing stability information in Galleon configuration elements at a higher granularity level than a feature-spec (i.e. configs, layers, feature-groups).

Reply:
@bstansberry , thanks, I added your suggestion in the Non Requirements